src/utils/compressImages.ts

```diff
@@ -0,0 +1,131 @@
+import { ImageCompressionOptions } from '../lib/Sendbird';
+import pxToNumber from './pxToNumber';
+import { Logger } from '../lib/SendbirdState';
+
+interface CompressImageProps {
+  imageFile: File;
+  compressionRate: number;
+  resizingWidth?: number;
+  resizingHeight?: number;
+}
+
+const compressImage = ({
+  imageFile,
+  compressionRate,
+  resizingWidth,
+  resizingHeight,
+}: CompressImageProps): Promise<File> => {
+  const image = document.createElement('img');
+  return new Promise((resolve, reject) => {
+    image.src = URL.createObjectURL(imageFile);
+    image.onerror = reject;
+    image.onload = () => {
+      URL.revokeObjectURL(image.src);
+      const canvas = document.createElement('canvas');
+
+      const originWidth = image.width;
+      const originHeight = image.height;
+      const widthRatio = originWidth / (resizingWidth || originWidth);
+      const heightRatio = originHeight / (resizingHeight || originHeight);
+
+      let targetResizingWidth = resizingWidth || originWidth;
+      let targetResizingHeight = resizingHeight || originHeight;
+
+      // Use the more impactful value, so the original images' ratio won't be broken
+      if (widthRatio > heightRatio) {
+        targetResizingHeight = originHeight / (resizingWidth ? widthRatio : 1);
+      } else if (heightRatio > widthRatio) {
+        targetResizingWidth = originWidth / (resizingHeight ? heightRatio : 1);
+      }
+
+      canvas.width = targetResizingWidth;
+      canvas.height = targetResizingHeight;
+
+      const ctx = canvas.getContext('2d');
+      if (!ctx) {
+        reject(new Error('Failed to get canvas 2d context'));
+        return;
+      }
+
+      ctx.drawImage(image, 0, 0, targetResizingWidth, targetResizingHeight);
+
+      ctx.canvas.toBlob(
+        (blob) => {
+          if (blob) {
+            const file = new File([blob], imageFile.name, { type: imageFile.type });
+            resolve(file);
+          } else {
+            reject(new Error('Failed to compress image'));
+          }
+        },
+        imageFile.type,
+        compressionRate,
+      );
+    };
+  });
+};
+
+export interface CompressImagesProps {
+  files: File[];
+  imageCompression: ImageCompressionOptions;
+  logger?: Logger;
+}
+export const compressImages = async ({
+  files,
+  logger,
+  imageCompression,
+}: CompressImagesProps) => {
+  const { compressionRate } = imageCompression;
+  const resizingWidth = pxToNumber(imageCompression.resizingWidth);
+  const resizingHeight = pxToNumber(imageCompression.resizingHeight);
+
+  if (!(Array.isArray(files) && files.length > 0)) {
+    logger.warning('useImageCompression: There is no file.', files);
+    return;
+  }
+  if (compressionRate < 0 || 1 < compressionRate) {
+    logger.warning('useImageCompression: The compressionRate is not acceptable.', compressionRate);
+    return;
+  }
+
+  const compressedImages: File[] = [];
+  const notCompressedFiles: File[] = [];
```

Comment:
Do we have a reason not to keep consistency in the terms `images` <-> `files`? If not, why don't we use simpler name here; `succeed & failed`
```suggestion
  const succeedFiles: File[] = [];
  const failedFiles: File[] = [];
```

Reply:
I changed the return values, pls check it again